plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java

```diff
@@ -462,24 +462,39 @@ private BrokerResponseNative submitBrokerQueryJson(ConnectorSession session, Pin
     public Iterator<BrokerResultRow> createResultIterator(ConnectorSession session, PinotQuery query, List<PinotColumnHandle> columnHandles)
     {
         BrokerResponseNative response = submitBrokerQueryJson(session, query);
-        return fromResultTable(response.getResultTable(), columnHandles);
+        return fromResultTable(response, columnHandles);
     }
 
     @VisibleForTesting
-    public static ResultsIterator fromResultTable(ResultTable resultTable, List<PinotColumnHandle> columnHandles)
+    public static ResultsIterator fromResultTable(BrokerResponseNative brokerResponse, List<PinotColumnHandle> columnHandles)
     {
-        requireNonNull(resultTable, "resultTable is null");
+        requireNonNull(brokerResponse, "brokerResponse is null");
         requireNonNull(columnHandles, "columnHandles is null");
+        ResultTable resultTable = brokerResponse.getResultTable();
         String[] columnNames = resultTable.getDataSchema().getColumnNames();
         Map<String, Integer> columnIndices = IntStream.range(0, columnNames.length)
                 .boxed()
                 // Pinot lower cases column names which use aggregate functions, ex. min(my_Col) becomes min(my_col)
                 .collect(toImmutableMap(i -> columnNames[i].toLowerCase(ENGLISH), identity()));
         int[] indices = new int[columnNames.length];
+        int[] inverseIndices = new int[columnNames.length];
         for (int i = 0; i < columnHandles.size(); i++) {
             indices[i] = columnIndices.get(columnHandles.get(i).getColumnName().toLowerCase(ENGLISH));
+            inverseIndices[indices[i]] = i;
         }
-        return new ResultsIterator(resultTable, indices);
+        List<Object[]> rows = resultTable.getRows();
+        // If returning from an aggregate with no grouping columns, make sure all non-count columns are null
+        if (brokerResponse.getNumDocsScanned() == 0 && resultTable.getRows().size() == 1) {
```

Comment:
```
// If returning from a global aggregation (no grouping columns) over an empty table, NULL-out all aggregation function results except for `count()`
```

Some questions

- Is `brokerResponse.getNumDocsScanned() == 0` equivalent to saying "table is empty"? Can a document hold no rows? What if there were documents scanned, but data was filtered out (`SELECT max(x) FROM non_empty_table WHERE y = non_existent_value`)?
- `resultTable.getRows().size() == 1`  combined with "no data" is a tricky way saying "global aggregation". Would checking `query.getGroupByClauses()==0` be sufficient? (assuming `getGroupByClauses` is populated correctly, which i didn't check)

Reply:
The `numDocsScanned` in the response will be set to 0 if the result set is empty even if the dataset before the filter is non-empty. The `numSegmentsScanned` in the response will be > 0 if the data set was not empty, i.e. it scanned over rows but the filter removed them.

Good point: I added groupingColumns as an extra verification.

There is a test to verify this behavior in `testAggregationPushdown` - the one with the comment: `// Test single row from pinot where filter results in an empty result set.`